executor.go

```diff
@@ -397,7 +397,10 @@ func doesFragmentConditionMatch(eCtx *ExecutionContext, fragment ast.Node, ttype
 		if conditionalType == ttype {
 			return true
 		}
-
+                if conditionalType.Name() == ttype.Name() {
```

Comment:
Thanks for clarification @pyros2097 - let's keep this since you've provide a handy use case.